File: scripts/export_products.js
const fs = require('fs');
const path = require('path');

// Path to frontend mock file
const mockPath = path.join(__dirname, '..', 'frontend', 'src', 'data', 'mockProducts.ts');
const content = fs.readFileSync(mockPath, 'utf8');

// Extract the array between the brackets
const start = content.indexOf('[');
const end = content.lastIndexOf(']') + 1;
let arrayStr = content.substring(start, end);

// Replace 'baseImage' calls with actual URLs
arrayStr = arrayStr.replace(/baseImage\(['"]([^'"]+)['"]\)/g, (match, seed) => `"https://picsum.photos/seed/${seed}/400/400"`);

// Parse as JavaScript (we need to evaluate it, but it's easier to write a temporary file)
// We'll use a safer approach: write a temporary JSON file.
// Since it's TypeScript, we need to remove type annotations.
// Quick and dirty: we'll use eval with a function.
// But to avoid security issues, we'll just create a JSON file manually from the mock data.

console.log('Please manually copy the mockProducts array to backend/data/products.json');
console.log('Or run a Python script to convert it.');

File: scripts/export_products.py
import re
import json
from pathlib import Path

ts_file = Path(__file__).parent.parent / "frontend" / "src" / "data" / "mockProducts.ts"
if not ts_file.exists():
    print("Frontend mock file not found")
    exit(1)

content = ts_file.read_text(encoding="utf-8")
# Remove TypeScript type annotations and export
# We'll extract the array by finding the content between 'export const mockProducts: Product[] = ' and ';'
match = re.search(r'export const mockProducts: Product\[\] = (\[[\s\S]*?\]);', content)
if not match:
    print("Could not find mockProducts array")
    exit(1)

array_text = match.group(1)

# Replace baseImage calls with string literals
def replace_baseimage(match):
    seed = match.group(1)
    return f'"https://picsum.photos/seed/{seed}/400/400"'

array_text = re.sub(r'baseImage\(["\']([^"\']+)["\']\)', replace_baseimage, array_text)

# Now we have a string that looks like a JavaScript array.
# We'll evaluate it as Python using ast.literal_eval? Not safe.
# We'll use json5 or simple eval with caution.
# Simpler: write it to a temporary .js file and run node to get JSON.
# But we can just replace single quotes with double quotes and hope.
# Let's do a quick and dirty conversion.

# We'll use a trick: replace ' with " for keys and strings, but careful with nested.
# Better: use the json module with strict=False? Not possible.
# We'll just rely on the fact that it's valid JS and use a Node script.

print("Please run the Node script export_products.js instead.")